### PR TITLE
[ENG-1923][ENG-1849] Fix sparse node relationships

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

`spare/nodes/<node_id>/children` actually points to `spare/nodes/<node_id>/` without breaking unit tests. 👎 

## Changes

- fix sparse node children regex that was breaking the route
- shuffle tests into their own directory and add new tests for sparse node children.
- generalizes SparseNodeLinkedRegistrationsList to work for registrations too.

## QA Notes

Re-test everything having to do with sparse nodes again. It seems like we potentially have test gaps. Test permissions, filtering etc on sparse list and detail endpoints and their relationships. 

## Documentation

🐞 fix, no docs.

## Side Effects

I split out a lot of tests in order to write new ones better hopefully I didn't drop anything.

## Ticket

https://openscience.atlassian.net/browse/ENG-1849
https://openscience.atlassian.net/browse/ENG-1923

### CODE REVIEW

File: api/sparse/urls.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
         name=views.SparseNodeList.view_name,
     ),
     url(
-        r'^nodes/(?P<node_id>\w+)/',
+        r'^nodes/(?P<node_id>\w+)/$',
```
Comment (Johnetordoff) [inline api/sparse/urls.py:14]: This and the route change on ln 40 are the only truly necessary changes for this ticket, the rest is just to write more organized tests.

File: api_tests/nodes/views/test_node_detail.py [hunk 2]
```diff
@@ -702,10 +702,6 @@ def wiki_private(self, user, project_private):
     def url_public(self, project_public):
         return '/{}nodes/{}/'.format(API_BASE, project_public._id)
 
-    @pytest.fixture()
-    def sparse_url_public(self, project_public):
```
Comment (Johnetordoff) [inline api_tests/nodes/views/test_node_detail.py:706]: Anything that's sparse node related is moved to it's own test directory.

File: api_tests/nodes/views/test_node_list.py [hunk 3]
```diff
@@ -175,9 +167,7 @@ def test_node_list_has_proper_root(self, app, user, url, sparse_url, is_sparse):
             project = AbstractNode.load(project_json['id'])
             assert project_json['embeds']['root']['data']['id'] == project.root._id
 
-    @pytest.mark.parametrize('is_sparse', [True, False])
-    def test_node_list_sorting(self, app, url, sparse_url, is_sparse):
-        url = sparse_url if is_sparse else url
+    def test_node_list_sorting(self, app, url):
```
Comment (Johnetordoff) [inline api_tests/nodes/views/test_node_list.py:170]: I de-parametrized this and moved the other half to sparse node.

File: admin/spam/views.py [hunk 4]
```diff
@@ -107,32 +106,27 @@ def get_context_data(self, **kwargs):
...
-            raise Http404('Spam with id "{}" not found.'.format(spam_id))
-        update_admin_log(
-            user_id=self.request.user.id,
-            object_id=spam_id,
-            object_repr='Comment',
-            message=log_message,
-            action_flag=log_action
-        )
-        return super(SpamDetail, self).form_valid(form)
+        item = Comment.load(self.kwargs.get('spam_id'))
```
Comment (brianjgeiger) [inline admin/spam/views.py:109]: Almost there. Just need to get this back out of the PR.

**Pull request merged**